File: kitalalos_jatek_hibakorlattal/Pas/eva.pas
unit eva;
interface
  function GetN:longint;
  function GetH:integer;
  function Kerdes(A,B:longint):integer;
  procedure Megoldas(x:longint);
implementation
  const
    MaxH=14;
    maxM=4000000;
    Init:boolean=true;
  var
    N,H:longint;
    bal,jobb:longint;
    Kerd:longint;
	
  procedure Vege(ki:longint; msg:String);
  begin{Vege}
	writeln(msg);
    Halt;
  end{Vege};

procedure Elofel;
begin{Elofel}
  readln(N, H);
  bal:=1; jobb:=N;
  kerd:=0;
  init:=false;
end{Elofel};

  function GetN:longint;
  begin
    if Init then begin
      EloFel;
      Init:=false;
    end;
    GetN:=N;
  end{GetN};
  
  function GetH:integer;
  begin
    if Init then begin
       EloFel;
      Init:=false;
    end;
    GetH:=H;
  end{GetH};

  function Kerdes(A,B:longint):integer;
  var v:integer ;
  begin{Kerdes}
    if Init then begin
      Vege(0,'hiba, először GetN-t kell hívni');
    end;
    if (A<1)or(A>N)or(B<1)or(B>N)or(A>B) then begin
      Vege(-1,'hiba, érvénytelen paraméter Kerdes-ben');
    end;
	if A<bal then A:=bal;
    if B>jobb then B:=jobb;
    v:=0;
    bal:=A; jobb:=B;
    inc(kerd);
    Kerdes:=v;
  end{Kerdes};

  procedure Megoldas(x:longint);
  begin{Megoldas}
    if Init then begin
      vege(0, 'hiba, először GetN-t kell hívni');
    end;
    if (bal<>jobb) then 
		vege(0, 'hiba, blöfföltél');
	if (x<>bal) then
       Vege(0,'hiba, rossz válasz');
    vege(1, 'helyes');
  end{Megoldas};
begin

end.

File: kitalalos_jatek_hibakorlattal/C/eva.c
#include <stdio.h>
#include <stdlib.h>

#include "eva.h"

#define MaxH            15
#define MaxM            4000001L
#define true 1
#define false 0

int Init = true;
long _N, _H, _M, _A;
long _L[MaxH][MaxM];
long _bal, _jobb, _mH, _Kerd, _optM;

void Vege(int ki, char *msg)

{
	printf("%s\n", msg);
	exit(1-ki);
}  /*Vege*/

void Elofel() {
	long seed=0;
	scanf("%ld%ld%ld%*[^\n]", &_N, &_H, &seed);
	srand(seed);
	_A=rand()%_N+1;
	_bal = 1;
	_jobb = _N;
	_Kerd = 0;
	Init = false;
}  /*Elofel*/

long GetN() {
	if (Init) {
		Elofel();
		Init = false;
	}
	return _N;
}  /*GetN*/

int GetH() {
	if (Init) {
		Elofel();
		Init = false;
	}
	return _H;
}  /*GetH*/

int Kerdes(long A, long B) {
	long v;

	if (Init) {
		Vege(0, "hiba, el?ször GetN-t kell hívni");
	}
	if (A < 1 || A > _N || B < 1 || B > _N || A > B) {
		Vege(0, "hiba, érvénytelen paraméter Kerdes-ben");
	}

	
	if(_A<A) {
		return -1;
	}else if(A<=_A && _A<=B) {
		return 0;
	}else {
		return 1;
	}
	_Kerd++;
	return v;
}  /*Kerdes*/

void Megoldas(long x) {
	if (Init) {
		Vege(0, "hiba, el?ször GetN-t kell hívni");
	}

	if (x != _A)
		Vege(0, "hiba, rossz válasz");
	Vege(1, "helyes");
}  /*Megoldas*/
